### PR TITLE
Linting Updates from 55places-Facade

### PR DESCRIPTION
**Pull request by zrosenbauer**
- Valid-jsdoc for docblock linting
- update sort-comp to match up with component/prop preferred order

### CODE REVIEW

File: base/rules/best-practices.js [hunk 1]
```diff
@@ -215,6 +215,22 @@ module.exports = {
     // http://eslint.org/docs/rules/vars-on-top
     'vars-on-top': 2,
```
Comment (zrosenbauer) [inline base/rules/best-practices.js:217]: This is up for debate. Let me know if we want to change this.

File: base/rules/best-practices.js [hunk 2]
```diff
@@ -215,6 +215,22 @@ module.exports = {
...
+      prefer: {
+        'return': 'returns',
+      },
+      preferType: {
+        'object': 'Object',
+        'number': 'number',
+        'string': 'string',
+        'function': 'Function',
+        'array': 'Array'
+      },
```
Comment (cbourdage) [inline base/rules/best-practices.js:228]: Why are some capitalized and others not? I would suggest we just go with this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures

File: base/rules/style.js [hunk 3]
```diff
@@ -222,7 +222,7 @@ module.exports = {
 
     // require or disallow space before function opening parenthesis
     // http://eslint.org/docs/rules/space-before-function-paren
-    'space-before-function-paren': [2, { anonymous: 'always', named: 'never' }],
+    'space-before-function-paren': [2, { anonymous: 'never', named: 'never' }],
```
Comment (cbourdage) [inline base/rules/style.js:225]: Thank you.

**Review by cbourdage — CHANGES_REQUESTED** [review]

File: base/rules/best-practices.js (hunk 2 above)
Comment (zrosenbauer) [reply base/rules/best-practices.js:228]: Yea I wasn't sure on this one. @jrubins what do you think makes most sense?

Comment (zrosenbauer) [inline base/rules/best-practices.js]: @cbourdage so everything is capitalized?

**Comment by zrosenbauer** [comment]
Updated @cbourdage

File: base/rules/best-practices.js (hunk 1 above)
Comment (jrubins) [reply base/rules/best-practices.js:217]: I mean I think we should leave this. We shouldn't encounter this often though since we prefer `const` and `let`.

**Review by jrubins — APPROVED** [review]

File: base/rules/best-practices.js (hunk 2 above)
Comment (jrubins) [reply base/rules/best-practices.js:228]: Yes, I like to capitalize.

File: base/rules/best-practices.js (hunk 1 above)
Comment (zrosenbauer) [reply base/rules/best-practices.js:217]: +1

**Comment by zrosenbauer** [comment]
@cbourdage ping

**Review by cbourdage — APPROVED** [review]

**Pull request merged**